### PR TITLE
fix: moving collateral whitelist to BondMediator

### PR DESCRIPTION
**Pull request by CjHare**
### Purpose for this PR
Second part of the refactoring to implement https://github.com/windranger-io/windranger-treasury/issues/161

By moving collateral check into the mediator, the factor is now, or can be DAO ID agnostic.

### CODE REVIEW

**Comment by CjHare** [comment]
@aodhgan this change will impact your PR https://github.com/windranger-io/windranger-treasury/pull/208

Take a look, if it's okay, feel free to merge it 🙂

File: contracts/bond/BondMediator.sol [hunk 1]
```diff
@@ -35,31 +37,35 @@ contract BondMediator is
...
      * @param manager A deployed BondCurator contract to register created bonds with,
      * @param erc20CapableTreasury Treasury that receives forfeited collateral. Must not be address zero.
+     * @param erc20CollateralTokens Collateral token contract. Must not be address zero.
      */
     function initialize(
         address factory,
         address manager,
-        address erc20CapableTreasury
+        address erc20CapableTreasury,
+        address erc20CollateralTokens
```
Comment (aodhgan) [inline contracts/bond/BondMediator.sol:46]: should this be singular? erc20CollateralToken?

Comment (CjHare) [reply contracts/bond/BondMediator.sol:46]: Good question, it depends on the context. 
When passing around the address or IERC20 (or equivalent) the reference is to the token contract, which being the manager of the token, is plural.

This naming convention also carries over to the tests, where the variable name for the IERC20 contracts are plural 😁 

There's a number of OS cases, the one that springs to mind (and quickly found 😉 )  
https://github.com/jbx-protocol/juice-contracts-v2/blob/fd289805dbca5cc8c5677ef92e5461fe57f928d7/contracts/JBETHPaymentTerminal.sol#L246

**Review by aodhgan — APPROVED** [review]
lgtm

**Pull request merged**